### PR TITLE
Improve HttpEventAdaptor logs 

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
In the current implementation of the adapter, response codes are not handled and due to that runtime errors can not be troubleshooted. Hence we need to improve the current adapter to log detailed errors for troubleshooting purposes .

### CODE REVIEW

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java [hunk 1]
```diff
@@ -330,7 +330,15 @@ public void run() {
...
                 }
 
-                this.getHttpClient().executeMethod(hostConfiguration, method);
+                int responseCode = this.getHttpClient().executeMethod(hostConfiguration, method);
+                if (responseCode == 200 || responseCode == 201 || responseCode ==200) {
+                    log.info("Successfully connect to the endpoint. Relieved HTTP response code is : " +
+                            responseCode);
+                }
+                else {
+                    log.error("Error while connecting the endpoint. Relieved HTTP response code is : " +
```
Comment (wenodpathirana) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java:339]: better if we can log the error details in the reponse, you can access them from `method` object.

**Review by wenodpathirana — APPROVED** [review]

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapterFactory.java [hunk 2]
```diff
@@ -125,7 +125,10 @@ private boolean isFormUrlEncodedEnabled() {
 
         Map<String, String> globalProperties = EventAdapterUtil.
                 getGlobalProperties(HTTPEventAdapterConstants.ADAPTER_TYPE_HTTP);
-        return Boolean
-                .parseBoolean(globalProperties.get(HTTPEventAdapterConstants.ENABLE_FORM_URL_ENCODED));
+        if (globalProperties != null) {
+            return Boolean
+                    .parseBoolean(globalProperties.get(HTTPEventAdapterConstants.ENABLE_FORM_URL_ENCODED));
+        }
+        return false;
```
[the remark this answers is not in the document]
Comment (ChanikaRuchini) [reply components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapterFactory.java]: fixed

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java [hunk 3]
```diff
@@ -330,8 +330,16 @@ public void run() {
                     }
                 }
 
-                this.getHttpClient().executeMethod(hostConfiguration, method);
-
+                int responseCode = this.getHttpClient().executeMethod(hostConfiguration, method);
+                if (responseCode / 100 == 2) {
+                    if (log.isDebugEnabled()) {
+                        log.info("Successfully connected to the endpoint. Received HTTP response code is : " +
```
Comment (AnuGayan) [inline components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java:336]: Shall we make this debug?

**Review by AnuGayan — CHANGES_REQUESTED** [review]

File: components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java (hunk 3 above)
Comment (ChanikaRuchini) [reply components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.http/src/main/java/org/wso2/carbon/event/output/adapter/http/HTTPEventAdapter.java]: Fixed

**Review by AnuGayan — APPROVED** [review]

**Pull request merged**